src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -422,6 +422,19 @@ private void PublishCode()
 #endif
                 );
 
+            if (_usedFrameInfos != _numFrameInfos)
+            {
+                // jitNativeCode might retry itself when exception happens inside jit.
```

Comment:
Would it be better to clear the already allocated frameinfos when jitNativeCode retries?

Reply:
Yes, you are right, it would be a better solution. At first I did not see existing jit interface to do this, but for some reason missed `reportFatalError`. I've checked that cleanup in `reportFatalError` works, so I'll update this PR.